win32ss/gdi/ntgdi/freetype.c

```diff
@@ -1577,16 +1599,36 @@ IntGdiAddFontResourceEx(PUNICODE_STRING FileName, DWORD Characteristics,
     PVOID Buffer = NULL;
     IO_STATUS_BLOCK Iosb;
     PVOID SectionObject;
-    SIZE_T ViewSize = 0;
+    SIZE_T ViewSize = 0, Length;
     LARGE_INTEGER SectionSize;
     OBJECT_ATTRIBUTES ObjectAttributes;
     GDI_LOAD_FONT LoadFont;
     INT FontCount;
     HANDLE KeyHandle;
+    UNICODE_STRING PathName;
+    WCHAR szPath[MAX_PATH];
     static const UNICODE_STRING TrueTypePostfix = RTL_CONSTANT_STRING(L" (TrueType)");
 
+    /* Build PathName */
+    Length = wcslen(SharedUserData->NtSystemRoot);
+    RtlStringCbCopyNW(szPath, sizeof(szPath), FileName->Buffer,
+                      min(Length * sizeof(WCHAR), FileName->Length));
+    if (FileName->Length > Length * sizeof(WCHAR) &&
+        _wcsnicmp(szPath, SharedUserData->NtSystemRoot, Length) == 0)
+    {
+        RtlStringCbCopyW(szPath, sizeof(szPath), L"\\SystemRoot");
+        RtlStringCbCatNW(szPath, sizeof(szPath), &FileName->Buffer[Length],
+                         FileName->Length - Length * sizeof(WCHAR));
+        RtlInitUnicodeString(&PathName, szPath);
```

Comment:
Using UNICODE_STRING functions here seems both more efficient and (to me, at least) more readable. I.e. `RtlInitEmptyUnicodeString` + `RtlAppendUnicodeToString` + `RtlAppendUnicodeStringToString`

Reply:
Optimized.